doc.rkt

```diff
@@ -242,60 +250,68 @@
                      ;; position. If we were to instead call `indent-line!` first and then
                      ;; `remove-trailing-space!` second, the remove step could result in
                      ;; losing user entered code.
-                     (list (remove-trailing-space! mut-doc-text skip-this-line? line)
-                           (indent-line! mut-doc-text indenter line #:on-type? on-type?)))
+                     (list (if (false? (FormattingOptions-trimTrailingWhitespace fo))
+                               #f
+                               (remove-trailing-space! mut-doc-text skip-this-line? line))
```

Comment:
I think
```racket
(and (FormattingOptions-trimTrailingWhitespace fo)
     (remove-trailing-space! mut-doc-text skip-this-line? line))
```
will work

Reply:
No. `FormattingOptions-trimTrailingWhitespace` could be omited in the message and will be an `undef-object` which is not `#f`.

Reply:
If we assign `#f` as it's default value?

Reply:
Ok, then do we have chance to make it be `#f` when it's undefined?

Reply:
No. It naturally has three different states: undefined, boolean true, boolean false.
A trick to merge two states just for simpler syntactic sugar expression instead of simpler logic feels not good.

Reply:
Ok, I get it. You can mark this as resolved now